hooks/shapediver/viewer/interaction/utils/patternUtils.ts

```diff
@@ -21,6 +21,47 @@ import { IOutputApi, ITreeNode, OutputApiData } from "@shapediver/viewer";
  */
 export type NameFilterPattern = { [key: string]: string[][] };
 
+/**
+ * The black list of node names that should be ignored.
+ * 
+ * These node names will be ignored when checking the node names.
+ * These names are used in the ShapeDiver GH plugin for certain operations.
+ */
+const nodeNameBlackList = ["TransformZUpToYUp", "no_transformations"];
+
+/**
+ * Check if the node matches the pattern and add interaction data if it does.
+ * 
+ * For each pattern, the function will check if the node name matches the pattern.
+ * If the node name matches the pattern, the function will check the children of the node.
+ * Only if the node name matches the last pattern, the interaction data will be added.
+ * 
+ * @param node The node to check.
+ * @param pattern The pattern to check.
+ * @param count The current count of the pattern.
+ * @param result The result array. 
+ */
+export const gatherNodesForPattern = (node: ITreeNode, pattern: string[], count: number, result: ITreeNode[] = []): void => {
+	// if there is no original name or the node name is in the black list, ignore the node
+	if (!node.originalName || nodeNameBlackList.includes(node.originalName)) {
+		if (count !== 0) return;
+
+		for (const child of node.children) {
+			gatherNodesForPattern(child, pattern, 0, result);
+		}
```

Comment:
@MajorMeerkatThe3rd this special logic for count === 0 is not clear to me, please explain it in the code documentation